### ISSUE TITLE
Help with Parameters (-fb, -c) for running SLAMDUNK; comparison of published data

### ISSUE THREAD

**Issue by crueckle** [t-neumann/slamdunk#102]
Hi Tobias,

thank you for this great tool and the detailed documentation! 

To find out which parameters fit best to my analysis, I started to reanalyze some data to understand which parameters were chosen for the original publication and how I can then process my new data for comparison. 

When setting the parameters as suggested in Matsushima 2019., I get different results in the count table (analyizing e.g. SRR5678869.fastq) than the provided in GEO. See first lines of the count table results:

**Published GEO count table**

#slamdunk v0.2.4	3	sample info:	pulse_0h_R1_34330	1	pulse	0
#annotation:	170614custom_mESCannotation_counting_Thomas.bed	5a451868b8f3be84e2040f5872bce883
Chromosome	Start	End	Name	Length	Strand	ConversionRate	ReadsCPM	Tcontent	CoverageOnTs	ConversionsOnTs	ReadCount	TcReadCount	multimapCount	ConversionRateLower	ConversionRateUpper
chr18	38262373	38262629	0610009O20Rik	256	+	0.00257272907184	33.8718140325	61	5053	13	487	13	0	-1.0	-1.0
chr2	175656519	175656769	0610010B08Rik	250	+	0.0	0.417311877197	72	52	0	6	0	6	-1.0	-1.0
chr2	175754078	175754328	0610010B08Rik	250	+	0.0	0.278207918131	72	30	0	4	0	4	-1.0	-1.0
chr6	72352897	72353160	0610030E20Rik	263	+	0.00150602409639	3.89491085384	77	664	1	56	1	1	-1.0	-1.0
chr4	107896865	107897117	0610037L13Rik	252	+	0.000277932184547	31.3679427693	59	3598	1	451	1	2	-1.0	-1.0
chr4	107897552	107897803	0610037L13Rik	251	+	0.0019267822736	3.61670293571	69	519	1	52	1	0	-1.0	-1.0
chr8	77518328	77518578	0610038B21Rik	250	+	0.0	0.139103959066	75	15	0	2	0	0	-1.0	-1.0

**Running as suggested in Matsushima 2019:**
`SLAMDUNK all -r $REF -b GSE99970_GSE99970_mESC_counting_windows_1.bed -o ./ -t 5 -5 12 -n 100 -m -mv 0.2 -c
2 -rl 100 SRR5678869.fastq.gz
`
Chromosome	Start	End	Name	Length	Strand	ConversionRate	ReadsCPM	Tcontent	CoverageOnTs	ConversionsOnTs	ReadCount	TcReadCount	multimapCount	ConversionRateLower	ConversionRateUpper
chr18	38262373	38262629	.	256	+	0.0	32.70567757802839	61	5064	0	487	0	0	-1.0	-1.0
chr2	175656519	175656769	.	250	+	0.0	0.6715744882552032	72	82	0	10	0	10	-1.0	-1.0
chr2	175754078	175754328	.	250	+	0.0	0.13431489765104063	72	18	0	2	0	2	-1.0	-1.0
chr6	72352897	72353160	.	263	+	0.0	3.693659685403617	77	658	0	55	0	0	-1.0	-1.0
chr4	107896865	107897117	.	252	+	0.0	29.95222217618206	59	3573	0	446	0	1	-1.0	-1.0
chr4	107897552	107897803	.	251	+	0.0	3.492187338927056	69	522	0	52	0	0	-1.0	-1.0
chr8	77518328	77518578	.	250	+	0.0	0.13431489765104063	75	15	0	2	0	0	-1.0	-1.0

Running differently, included -fb as suggested in Herzog et al. 2017

`SLAMDUNK all -r $REF -b $UTR3 -o $PROJECT -t $THREADS -5 12 -n 100 -fb ../GSE99970_GSE99970_mESC_UTR_regions.bed -m -rl 100 SRR5678869.fastq.gz
`

Chromosome	Start	End	Name	Length	Strand	ConversionRate	ReadsCPM	Tcontent	CoverageOnTs	ConversionsOnTs	ReadCount	TcReadCount	multimapCount	ConversionRateLower	ConversionRateUpper
chr18	38262373	38262629	.	256	+	0.002764612954186414	32.70567757802839	61	5064	14	487	14	0	-1.0	-1.0
chr2	175656519	175656769	.	250	+	0.0	0.6715744882552032	72	82	0	10	0	10	-1.0	-1.0
chr2	175754078	175754328	.	250	+	0.0	0.13431489765104063	72	18	0	2	0	2	-1.0	-1.0
chr6	72352897	72353160	.	263	+	0.001519756838905775	3.693659685403617	77	658	1	55	1	0	-1.0	-1.0
chr4	107896865	107897117	.	252	+	0.000279876854184159	29.95222217618206	59	3573	1	446	1	1	-1.0	-1.0
chr4	107897552	107897803	.	251	+	0.0019157088122605363	3.492187338927056	69	522	1	52	1	0	-1.0	-1.0
chr8	77518328	77518578	.	250	+	0.0	0.13431489765104063	75	15	0	2	0	0	-1.0	-1.0

With these parameters, the result is closer to the published results. In recent publications, it seems that -fb is not used. My question would be whether you would recommend to use -fb or if it is better to stick to the suggested parameters as in Matsushima 2019 (no -fb; -c 2, -mv 0.2). I am wondering because for many UTRs it seems that with these parameters the calculated conversion rates are often 0. 

Thank you a lot in advance for your help.

**Comment by t-neumann** [comment]
Hi @crueckle 

Usually the -fb should not make things worse, at is is simply used to retain multimappers. For longer reads (>50bp) however, we found that multimapper retainment does not help a lot anymore, therefore you don't gain much with -fb. What is behind the $UTR3 parameter you sent? I cannot properly decipher what goes in there.

**Comment by crueckle** [comment]
Hi, thank you for your answer. 

The $UTR3 command is the path to GSE99970_GSE99970_mESC_counting_windows_1.bed. I use the same reference fasta and bed file with utr for all runs. I just wonder why the results are so different and especially why running like Matsushima 2019 leads for me to so different results.

**Comment by t-neumann** [comment]
I will read up on what he exactly run and get back to you.

**Comment by crueckle** [comment]
Thank you! I only was able to get close to the published count table when running SLAMDUNK with -fb.

**Comment by t-neumann** [comment]
Mhm so what I see is that you ommit the -c 2 parameter in the -fb version. What happens if you run this?

Running as suggested in Matsushima 2019:
SLAMDUNK all -r $REF -b GSE99970_GSE99970_mESC_counting_windows_1.bed -o ./ -t 5 -5 12 -n 100 -m -mv 0.2 -rl 100 SRR5678869.fastq.gz

So ommitting the -c 2 parameter?

**Comment by crueckle** [comment]
Yes, I tried that, but then the CPMs are still different from the published count table. Would you still recommend to run it like that?

#slamdunk v0.4.3	3	sample info:	SRR5678869.fastq	0	pulse	0
#annotation:	GSE99970_GSE99970_mESC_counting_windows_1.bed	5a451868b8f3be84e2040f5872bce883
Chromosome	Start	End	Name	Length	Strand	ConversionRate	ReadsCPM	Tcontent	CoverageOnTs	ConversionsOnTs	ReadCount	TcReadCount	multimapCount	ConversionRateLower	ConversionRateUpper
chr18	38262373	38262629	0610009O20Rik	256	+	0.002764612954186414	31.166857049018297	61	5064	14	487	14	0	-1.0	-1.0
chr2	175656519	175656769	0610010B08Rik	250	+	0.0	4.287842756230443	72	669	0	67	0	67	-1.0	-1.0
chr2	175754078	175754328	0610010B08Rik	250	+	0.0	5.311805205479505	72	1024	0	83	0	83	-1.0	-1.0
chr6	72352897	72353160	0610030E20Rik	263	+	0.001519756838905775	3.5198709192936475	77	658	1	55	1	0	-1.0	-1.0
chr4	107896865	107897117	0610037L13Rik	252	+	0.000279876854184159	28.54295327281758	59	3573	1	446	1	1	-1.0	-1.0
chr4	107897552	107897803	0610037L13Rik	251	+	0.0019157088122605363	3.327877960059449	69	522	1	52	1	0	-1.0	-1.0
chr8	77518328	77518578	0610038B21Rik	250	+	0.0	0.12799530615613264	75	15	0	2	0	0	-1.0	-1.0

**Comment by t-neumann** [comment]
Yes if total reads and tcreads are consistent with the old ones, I would not worry about the CPMs too much. Usually we anyways calculate all these measures (RPMu, RPM) ourselves and feed the counts into DESeq2

**Comment by crueckle** [comment]
The total reads are not very consistent (otherwise the CPM calculation would be more similar).
(When only looking at UTR/Line 3 for example, the read counts are in the published table 67; in my run 6.). I was planning on using the CPMs for filtering as suggested in Herzog et al., 2017 (filtered for CPM>5). 

Thank you a lot for your quick help. I will run SLAMSeq as suggested and see how to filter for expressed transcripts.

**Comment by t-neumann** [comment]
I would try actually this command:

```
SLAMDUNK all -r $REF -b $UTR3 -o $PROJECT -t $THREADS -5 12 -n 100 -fb ../GSE99970_GSE99970_mESC_UTR_regions.bed -m -rl 100 -mv 0.2 SRR5678869.fastq.gz
```

So including the mv parameter to call lower  allele frequency T>C SNPs.

**Comment by crueckle** [comment]
> Mhm so what I see is that you ommit the -c 2 parameter in the -fb version. What happens if you run this?
> 
> Running as suggested in Matsushima 2019:
> SLAMDUNK all -r $REF -b GSE99970_GSE99970_mESC_counting_windows_1.bed -o ./ -t 5 -5 12 -n 100 -m -mv 0.2 -rl 100 SRR5678869.fastq.gz
> 
> So ommitting the -c 2 parameter?

I run it as suggested by you here, including -mv 0.2 but omitting -c as you suggested. But I did not run it using -fb. So do you mean including -fb?

**Comment by crueckle** [comment]
Yes, so running it with -fb and -mv 0.2 brings be to my first post, as I get the same results as in the run posted there last. 

Chromosome	Start	End	Name	Length	Strand	ConversionRate	ReadsCPM	Tcontent	CoverageOnTs	ConversionsOnTs	ReadCount	TcReadCount	multimapCount	ConversionRateLower	ConversionRateUpper
chr18	38262373	38262629	.	256	+	0.002764612954186414	32.70567757802839	61	5064	14	487	14	0	-1.0	-1.0
chr2	175656519	175656769	.	250	+	0.0	0.6715744882552032	72	82	0	10	0	10	-1.0	-1.0
chr2	175754078	175754328	.	250	+	0.0	0.13431489765104063	72	18	0	2	0	2	-1.0	-1.0
chr6	72352897	72353160	.	263	+	0.001519756838905775	3.693659685403617	77	658	1	55	1	0	-1.0	-1.0

I just want to make sure I know how to best run it with my own data. As you suggested, I would probably run it as follows 

Running as suggested in Matsushima 2019, without -c:
SLAMDUNK all -r $REF -b GSE99970_GSE99970_mESC_counting_windows_1.bed -o ./ -t 5 -5 12 -n 100 -m -mv 0.2 -rl 100 SRR5678869.fastq.gz

**Comment by t-neumann** [comment]
Ok I tried to figured out the reproducibility - if you want a recommendation for your real dataset, what read length are you operating with?

**Comment by crueckle** [comment]
I do not have the data yet, but it will be 150 bp. Does more than -rl 100 change with that?

**Comment by t-neumann** [comment]
Yes that would have to be set to 150. Also, then I would go for more robust T>C read calling with -c 2. You would not need to rely as much on multimapper retainment with that read length, so I would use -n 1 and remove the -m flag.

What cell line / model organism are you working on?

**Comment by crueckle** [comment]
Thanks a lot! I will work in mESC. So I am thinking about using your published 3'UTR annotation.

**Comment by crueckle** [comment]
Just to understand, with 75 or 100 you would rather remove -c and put -n 75 and -m

**Comment by t-neumann** [comment]
yes because with shorter reads you have less chances to observe to T>C mutations so you will pick up less signal. It is not -n 75 but -rl 75 - the -rl parameter simply refers to the read length

**Comment by crueckle** [comment]
Yes, perfect, thank you!
